### PR TITLE
RZ spectral solver header to be called within PSATD ifdefs

### PR DESCRIPTION
**Pull request by RevathiJambunathan**
RZ was not compiling for FDTD as it was looking for `fftw3.h` since the spectral_solver header files for RZ were called outside of PSATD ifdefs.
I am not sure how the CI tests were passing. In particular, silver_mueller_rz.

### CODE REVIEW

**Comment by RevathiJambunathan** [comment]
@EZoni showed offline that the executable used in CI test is compiled with PSATD and that is why the `silver_mueller_rz_z` test was passing.

File: Source/Evolve/WarpXEvolve.cpp [hunk 1]
```diff
@@ -16,9 +16,9 @@
 #include "Evolve/WarpXDtType.H"
 #ifdef WARPX_USE_PSATD
 #   include "FieldSolver/SpectralSolver/SpectralSolver.H"
```
Comment (dpgrote) [inline Source/Evolve/WarpXEvolve.cpp]: I think this should include only one of these, so should be something like this
```
#ifdef WARPX_USE_PSATD
#    ifdef WARPX_DIM_RZ
#        include "FieldSolver/SpectralSolver/SpectralSolverRZ.H"
#    else
#        include "FieldSolver/SpectralSolver/SpectralSolver.H"
#    endif
#endif
```

File: Source/FieldSolver/WarpXPushFieldsEM.cpp [hunk 2]
```diff
@@ -11,12 +11,11 @@
 #include "BoundaryConditions/PML.H"
 #include "Evolve/WarpXDtType.H"
 #include "FieldSolver/FiniteDifferenceSolver/FiniteDifferenceSolver.H"
-#if defined(WARPX_DIM_RZ) || defined(WARPX_USE_PSATD)
+#if defined(WARPX_USE_PSATD)
```
Comment (dpgrote) [inline Source/FieldSolver/WarpXPushFieldsEM.cpp:14]: Same here as above, that only one should be imported. It looks like SpectralFieldData.H is Ok and should be imported with and without RZ.

File: Source/WarpX.cpp [hunk 3]
```diff
@@ -20,10 +20,10 @@
 #ifdef WARPX_USE_PSATD
 #   include "FieldSolver/SpectralSolver/SpectralSolver.H"
```
Comment (dpgrote) [inline Source/WarpX.cpp]: And the same here, only one should be imported. `SpectralKSpace.H` should be imported with and without RZ.

File: Source/FieldSolver/WarpXPushFieldsEM.cpp (hunk 2 above)
Comment (RevathiJambunathan) [reply Source/FieldSolver/WarpXPushFieldsEM.cpp:14]: Thank you @dpgrote 
This is fixed now

**Comment by ax3l** [comment]
Thanks for catching this!

Compile-time CI coverage added in #2048
X-Ref: regression of #1947

**Review by ax3l — APPROVED** [review]

**Pull request merged**